### PR TITLE
Add option to sign requests

### PR DESCRIPTION
**Pull request by DavidTanner**
signing

Some services are locked behind signed requests. OpenSearch can be set up with Role Based permissions, even if they are locked behind a VPC as well.  This just allows us to sign requests using roles.

### CODE REVIEW

**Comment by rmneidermyer** [comment]
I am probably overlooking this in the review, what is the option to use request signing?

**Comment by DavidTanner** [comment]
Signing requests allows the alpha client to make requests to services that use role based access. In my case I want to use `alpha-cli` to proxy the OpenSearch dashboards so we can make queries against the cluster.

https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html

File: src/alpha.ts [hunk 1]
```diff
@@ -84,4 +86,26 @@ export class Alpha extends Axios {
 
     return response as R;
   }
+
+  get<T = any, R = AxiosResponse<T>, D = any>(url: string, config?: AlphaOptions<D>): Promise<R> {
```
Comment (macsj200) [inline src/alpha.ts:90]: Is it necessary to explicitly define each of these methods?

Comment (DavidTanner) [reply src/alpha.ts:90]: Workout this the typing doesn't work to allow us to use `AlphaOptions` without casting. With this you can specify our custom options on a per request basis.

**Review by xdumaine — APPROVED** [review]

**Pull request merged**